internal/venice-common/src/main/java/com/linkedin/venice/init/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -0,0 +1,147 @@
+    if (response.isError()) {
+      if (STORE_NOT_FOUND.equals(response.getErrorType())) {
+        return null;
+      } else {
+        throw new VeniceException(response.getError());
+      }
+    }
+
+    return response.getCluster();
+  }
+
+  @Override
+  protected boolean doesStoreExist() {
+    StoreResponse response = controllerClient.retryableRequest(10, cc -> cc.getStore(getSchemaStoreName()));
+    if (response.isError()) {
+      if (STORE_NOT_FOUND.equals(response.getErrorType())) {
+        return false;
+      } else {
+        throw new VeniceException(response.getError());
+      }
+    }
+
+    return true;
+  }
+
+  @Override
+  protected boolean shouldCreateMissingSchemaStores() {
+    return autoCreateWriteSystemStores;
+  }
+
+  @Override
+  protected void createSchemaStore(Schema firstKeySchema, Schema firstValueSchema) {
+    NewStoreResponse newStoreResponse = controllerClient.createNewSystemStore(
+        systemStore.getZkSharedStoreName(),
+        VeniceConstants.SYSTEM_STORE_OWNER,
+        firstKeySchema.toString(),
+        firstValueSchema.toString());
+
+    if (newStoreResponse.isError()) {
+      throw new VeniceException(newStoreResponse.getError());
+    }
+  }
+
+  @Override
+  protected void updateStore(UpdateStoreQueryParams updateStoreQueryParams) {
+    controllerClient.updateStore(getSchemaStoreName(), updateStoreQueryParams);
+  }
+
+  @Override
+  protected SchemaEntry getRegisteredKeySchema() {
+    SchemaResponse response = controllerClient.getKeySchema(getSchemaStoreName());
+    return new SchemaEntry(response.getId(), AvroCompatibilityHelper.parse(response.getSchemaStr()));
+  }
+
+  @Override
+  protected Collection<SchemaEntry> getAllRegisteredValueSchemas() {
+    MultiSchemaResponse response = controllerClient.getAllValueSchema(getSchemaStoreName());
+
+    return Arrays.stream(response.getSchemas())
+        .map(schema -> new SchemaEntry(schema.getId(), AvroCompatibilityHelper.parse(schema.getSchemaStr())))
+        .collect(Collectors.toSet());
+  }
+
+  @Override
+  protected void registerValueSchema(Schema schema, int version) {
+    controllerClient.addValueSchema(getSchemaStoreName(), schema.toString(), version);
+  }
+
+  @Override
+  protected ValueAndDerivedSchemaId getDerivedSchemaInfo(Schema derivedSchema) {
+    SchemaResponse response =
+        controllerClient.getValueOrDerivedSchemaId(getSchemaStoreName(), derivedSchema.toString());
+    return new ValueAndDerivedSchemaId(response.getId(), response.getDerivedSchemaId());
+  }
+
+  @Override
+  protected void registerDerivedSchema(int valueSchemaVersion, Schema writeComputeSchema) {
+    controllerClient.addDerivedSchema(getSchemaStoreName(), valueSchemaVersion, writeComputeSchema.toString());
+  }
+
+  @Override
+  public void verify() {
+    // Wait for newly registered schema to propagate to local fabric
+    schemaPresenceChecker.verifySchemaVersionPresentOrExit();
```

Comment:
IIUC, in multi-colo mode, child colo standby controller and server will call parent controller to register new schema. But the verification here is checking schema in local child colo. If admin consumption is delayed in child colo, new schema will not be registered in child colo, the verification will fail, server will fail to start, right?

Reply:
I will be modifying this to wait for a while till this check succeeds. This would ensure that the schema is truly registered in local colo before we let the server startup continue

Reply:
We only register with child controller now, so, only checking local colo is sufficient